### PR TITLE
Firebase

### PR DESCRIPTION
**Pull request by faboweb**
Added firebase as the backend.
Refactored some parts of the application.

### CODE REVIEW

**Comment by nylira** [comment]
Looking awesome! I tried to run it and and found a couple of bugs so far.

1. Users are allowed to create new threads without signing in first. We need to require auth on this.
2. After signing in and leaving a comment, the comment is created with an empty name. Somehow the localstorage user isn't being populated correctly after signing in for the first time.

Looks like I have to refresh the pag first to fully log in.

**Comment by faboweb** [comment]
@nylira Thx for the feedback. On it.

File: src/store/modules/user.js [hunk 1]
```diff
@@ -54,13 +54,16 @@ const actions = {
...
+        if (!userFromDb || !userFromDb.uid) {
+          let newUser = {
             id: user.displayName.toLowerCase().replace(' ', '-'),
             uid: user.uid,
             name: user.displayName,
             email: user.email
-          })
+          }
+          userRef.set(newUser)
+          .then(() => resolve(newUser), reject)
```
Comment (faboweb) [inline src/store/modules/user.js]: resolving after the new user flow was missing

**Comment by faboweb** [comment]
@nylira found another security bug (changing votes of your posts)
All three issues are fixed now.

**Comment by nylira** [comment]
Let's merge this. I'll create new issues with anything I'll find but this is a good base to develop from. I'll push this live for our team to test. Good job.

**Pull request merged**